management/server/peer.go

```diff
@@ -220,10 +223,28 @@ func (am *DefaultAccountManager) GetNetworkMap(peerKey string) (*NetworkMap, err
 	}
 
 	var res []*Peer
-	for _, peer := range account.Peers {
-		// exclude original peer
-		if peer.Key != peerKey {
-			res = append(res, peer.Copy())
+	rules, err := am.Store.GetPeerSrcRules(account.Id, peerKey)
```

Comment:
fetching only rules where the peer is the source will give you an incomplete list of peers when the peer belongs only to a destination group